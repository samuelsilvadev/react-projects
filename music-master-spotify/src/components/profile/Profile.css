.profile {
    display: flex;
    justify-content: flex-start;
}

.profile__image {
    border-radius: 50%;
    object-fit: cover;
    height: 15rem;
    width: 15rem;
}

.profile__details {
    text-align: left;
    margin-left: 2rem;
}

.profile__genres {
    list-style: none;
    margin-bottom: 0;
    padding-left: 0;
}

.profile__genres>li {
    display: inline-block;
    padding-right: .3rem;
}