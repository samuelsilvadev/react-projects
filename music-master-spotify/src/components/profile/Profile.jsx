import React from 'react';
import PropTypes from 'prop-types';

import './Profile.css';

import texts from './../../translate';

class Profile extends React.Component {
	
	_renderGenres(genres) {
		return genres.map((genre, index) => <li key={index}>{` ${genre} ${genres[index + 1] ? '& ' : ''} `}</li>);
	}

	render() {
		const { artist } = this.props;

		const [ lgImage, mdImage, sImage, xsImage ] = artist.images;
		return (
			<section className="profile">
				<img className="profile__image" src={lgImage.url} alt={artist.name} />
				<div className="profile__details">
					<h2 className="profiles__name">{artist.name}</h2>
					<h3 className="profiles__followers">{artist.followers.total} { texts.followers }</h3>
					<ul className="profile__genres">{this._renderGenres(artist.genres)}</ul>
				</div>
			</section>
		);
	}
}

Profile.propTypes = {
	artist: PropTypes.shape({
		external_urls: PropTypes.shape({
			spotify: PropTypes.string
		}),
		followers: PropTypes.shape({
			href: PropTypes.string,
			total: PropTypes.number
		}),
		genres: PropTypes.array,
		href: PropTypes.string,
		id: PropTypes.string,
		images: PropTypes.array,
		name: PropTypes.string,
		popularity: PropTypes.number,
		type: PropTypes.string,
		url: PropTypes.string
	})
};

Profile.defaultProps = {
	artist: {
		external_urls: '',
		followers: {
			href: '',
			total: 0
		},
		genres: [],
		href: '',
		id: '',
		images: [],
		name: '',
		popularity: 0,
		type: '',
		url: ''
	}
};

export default Profile;
