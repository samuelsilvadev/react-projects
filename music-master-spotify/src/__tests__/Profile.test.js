import React from 'react';
import { shallow } from 'enzyme';

import Profile from '../components/profile/Profile';

describe('<Profile />', () => {
	it('should render correctly component', () => {
		const artist = {
			images: [{
                url: 'https://google.com',
                name: 'Test',
            }],
            followers: {
                total: 10,
            }
		};

		const tree = shallow(<Profile artist={artist}/>);

		expect(tree).toMatchSnapshot();
	});
});
