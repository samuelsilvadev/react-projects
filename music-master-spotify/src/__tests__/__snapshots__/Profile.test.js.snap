// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render correctly component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Profile
    artist={
      Object {
        "followers": Object {
          "total": 10,
        },
        "images": Array [
          Object {
            "name": "Test",
            "url": "https://google.com",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          className="profile__image"
          src="https://google.com"
        />,
        <div
          className="profile__details"
        >
          <h2
            className="profiles__name"
          />
          <h3
            className="profiles__followers"
          >
            10
             
            followers
          </h3>
          <ul
            className="profile__genres"
          />
        </div>,
      ],
      "className": "profile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "profile__image",
          "src": "https://google.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="profiles__name"
            />,
            <h3
              className="profiles__followers"
            >
              10
               
              followers
            </h3>,
            <ul
              className="profile__genres"
            />,
          ],
          "className": "profile__details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "profiles__name",
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                10,
                " ",
                "followers",
              ],
              "className": "profiles__followers",
            },
            "ref": null,
            "rendered": Array [
              10,
              " ",
              "followers",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "profile__genres",
            },
            "ref": null,
            "rendered": null,
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            className="profile__image"
            src="https://google.com"
          />,
          <div
            className="profile__details"
          >
            <h2
              className="profiles__name"
            />
            <h3
              className="profiles__followers"
            >
              10
               
              followers
            </h3>
            <ul
              className="profile__genres"
            />
          </div>,
        ],
        "className": "profile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "profile__image",
            "src": "https://google.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="profiles__name"
              />,
              <h3
                className="profiles__followers"
              >
                10
                 
                followers
              </h3>,
              <ul
                className="profile__genres"
              />,
            ],
            "className": "profile__details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "profiles__name",
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  10,
                  " ",
                  "followers",
                ],
                "className": "profiles__followers",
              },
              "ref": null,
              "rendered": Array [
                10,
                " ",
                "followers",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "profile__genres",
              },
              "ref": null,
              "rendered": null,
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
